src/platform/forms/save-in-progress/SaveInProgressIntro.jsx

```diff
@@ -89,25 +89,43 @@ class SaveInProgressIntro extends React.Component {
 
     if (login.currentlyLoggedIn) {
       if (savedForm) {
-        const lastUpdated =
-          savedForm.lastUpdated || _.get('metadata.lastUpdated', savedForm);
-        const savedAt = this.props.lastSavedDate
-          ? moment(this.props.lastSavedDate)
-          : moment.unix(lastUpdated);
-        const expiresAt = moment.unix(savedForm.metadata.expiresAt);
-        const expirationDate = expiresAt.format('MMMM D, YYYY');
-        const isExpired = expiresAt.isBefore();
+        /**
+         * lastSavedDate = JS time (ms) - always undefined?
+         * savedForms.lastUpdated = unix time (seconds)
+         * savedForms.metadata.expiresAt = unix time
+         * savedForms.metadata.lastUpdated = unix time
+         * savedForms.metadata.savedAt = JS time (ms)
+         */
+        const { metadata = {} } = savedForm;
+        const lastUpdated = savedForm.lastUpdated || metadata.lastUpdated;
+
+        let savedAt = '';
+        if (this.props.lastSavedDate) {
+          savedAt = new Date(this.props.lastSavedDate);
+        } else if (lastUpdated) {
+          savedAt = fromUnixTime(lastUpdated);
+        }
+
+        const expiresAt = fromUnixTime(savedForm.metadata.expiresAt);
+        const expirationDate = format(expiresAt, 'MMMM d, yyyy');
+        const isExpired = isBefore(expiresAt, new Date());
         const inProgressMessage = getInProgressMessage(formConfig);
 
         if (!isExpired) {
-          const lastSavedDateTime = savedAt.format('MMMM D, YYYY [at] h:mm a');
+          const lastSavedDateTime =
+            savedAt &&
+            `and was last saved on ${format(
+              savedAt,
+              "MMMM d, yyyy', at' h:mm aaaa z",
+            )}`;
+
           const H = `h${this.props.headingLevel}`;
           includesFormControls = true;
           alert = (
             <div className="usa-alert usa-alert-info background-color-only schemaform-sip-alert">
               <div className="schemaform-sip-alert-title">
                 <H className="usa-alert-heading vads-u-font-size--h3">
-                  {inProgressMessage} and was last saved on {lastSavedDateTime}
+                  {inProgressMessage} {lastSavedDateTime}
```

Comment:
I think I prefer `lastSavedDateTime` just being the date as opposed to also having "and was last saved on" in there.

Reply:
I changed it because some Cypress tests don't include a last saved time and as such that message would be "Your application is in progress and was last saved on" with no date. I don't know if this would happen with production data, but it'd be less confusing if it did.

Would you rather have me add that last part conditionally then?